---
title: "RNA-seq Analysis of variegatus RNAi"
author: "CiCi Zheng"
date: "6/7/2019"
output:
  html_document: default
---


This is the R Markdown documentation of the general RNA-seq differential expression analysis of M. luteus var. variegatus. It contains the analysis done using DESeq2 under the Bioconductor package in R, with the input being the "gene-read" txt file for individual samples. 

## Before the Analysis 
- The raw fastq data are from: "cooley_lab google drive/Transcriptome/Mlv_Mna_RNAseq_Libraries";
- Alignment is performed by Bowtie2 against C5_minlut.fasta;
- Read-count is done by HTSeq with C5_minlut.gtf;
- Identities of Sample 1-7 are: 
  M. naiandinus (S1), M. variegatus RNAi transformed (S2-S4), M. variegatus control (S5-S7)

## Differential Expression Analysis
- We first transform the ".txt" files to ".csv" then to a dataframe, we move S1 naiandinus to the last column to aid visualization: 
```{r, echo=FALSE}

samples <- list.files( pattern = ".txt")
for (i in 1:length(samples)) {
  sample=read.table(file=samples[i],header=F)
  write.csv(sample,file=paste(sub(".txt","",samples[i]),".csv") )
}

S1 = read.csv("S1 .csv")
S2 = read.csv("S2 .csv")
S3 = read.csv("S3 .csv")
S4 = read.csv("S4 .csv")
S5 = read.csv("S5 .csv")
S6 = read.csv("S6 .csv")
S7 = read.csv("S7 .csv")

# seq data: 
lut <- data.frame(S2$V2,S3$V2,S4$V2,S5$V2,S6$V2,S7$V2,S1$V2)
colnames(lut) <- c("Rvi1","Rvi2","Rvi3","Rc1","Rc2","Rc3","Nar")
rownames(lut) <- S1$V1
```
### Quality Assessment

- Plot the log2 transformation of raw read of all genes in one sample: 

Highly skewed data and zero heavy as expected. 

```{r ,echo=FALSE,include=TRUE, results="hide"}
library(ggplot2)
transformed_Nar <- log2(lut$Nar + 1)
ggplot(lut) +
  geom_histogram(aes(x = transformed_Nar), 
                 stat = "bin", bins = 200, colour = "white", fill = "#525252", binwidth = 0.6) +
  xlab("naiandinus")  + 
  ylab("expression(log[2](count + 1)")

```

- Create metadata for the experiment, we treat naiandinus as out-group:

```{r ,echo=FALSE}
# metadata
genotype <- c("var","var","var","var","var","var","narWT")
condition <- c("rnai","rnai","rnai","control","control","control","outgroup")
lut_meta <- data.frame(genotype,condition)
rownames(lut_meta) <- c("Rvi1","Rvi2","Rvi3","Rc1","Rc2","Rc3","Nar")
lut_meta
```

- Count normalization

Determine the size factors to use for normalization:
```{r, ,echo=FALSE}

library(DESeq2)
# Create DESeq object
lut_object <- DESeqDataSetFromMatrix(countData = lut,
                                    colData = lut_meta,
                                    design = ~condition)

# Determine the size factors to use for normalization
lut_object <- estimateSizeFactors(lut_object)
sizeFactors(lut_object)

```

Extract normalized counts and compute the correlation values between samples (diagonal value should be 1):
```{r ,echo=FALSE}

# Extract the normalized counts
lut_normalized_counts <- counts(lut_object, normalized=TRUE)

# Vst transformation: 
vsd_lut <- vst(lut_object, blind = TRUE)

# Extract the matrix of transformed counts
vsd_mat_lut <- assay(vsd_lut)

# Compute the correlation values between samples
vsd_cor_lut <- cor(vsd_mat_lut) 
head(vsd_cor_lut, n=3)
```

- Unsupervised clustering analyses
Hierarchical clustering with correlation heatmaps, the correalation between ingroups is kind of unclear:
```{r ,echo=FALSE, include=TRUE, results="hide"}
# Load library for pheatmap
library(pheatmap)

# Load library for tidyverse
library(tidyverse)

# Plot the heatmap, well at least nar stands out
pheatmap(vsd_cor_lut, annotation = select(lut_meta, condition))

```
Principal Component Analysis (PCA): outgroup seperate well with variegatus and is explained by PC1. Treatment samples seperate from control which is expalined by PC2. 

```{r, echo=FALSE}
# Plot the PCA of PC1 and PC2
plotPCA(vsd_lut, intgroup="condition")

```

## Differential Expression Analysis with DESeq2
- First we create DE object, and we need to make sure that our data fit the DESeq2 model
```{r, echo =FALSE}
# Run the DESeq2 analysis
lut_object <- DESeqDataSetFromMatrix(countData = lut,
                                     colData = lut_meta,
                                     design = ~condition)
deseq_lut <- DESeq(lut_object)
```

- variance increases with mean, and dispersion looks correct: which decreases with increasing mean and cluster around the maximum likelihood 
```{r, echo=FALSE, include=TRUE, results="hide"}
# Creating data frame with mean and variance for every gene
lut_matrix<- data.matrix(lut)
mean_count <- rowMeans(lut_matrix)
var_count <-rowVars(lut_matrix)
df <- data.frame(mean_count, var_count)
ggplot(df) +
  geom_point(aes(x=mean_count, y=var_count)) +
  scale_y_log10() +
  scale_x_log10() +
  xlab("Mean counts per gene") +
  ylab("Variance per gene")

# Dispersion looks good!
# decrease with increasing mean and cluster around the maximum likelihood 
plotDispEsts(deseq_lut)
```
- Result from the DESeq2 Negative Binomial Model

Summary gives us about ~650 significant genesdifferentially expressed genes: 
```{r, echo=FALSE}
# Get the DESeq2 contrasts:
lut_result <- results(deseq_lut, alpha = 0.05)
summary(lut_result)
```
The result dataframe looks like: 
```{r, echo=FALSE}
head(lut_result, n=3)
```

MA-plot before and after LFC shrinkage(accounting for artifacts):
```{r, echo=FALSE, include=TRUE, results="hide"}
# MA plot 
plotMA(lut_result, ylim=c(-8,8))
# LFC shrinkage: not a lot of differentially expressed genes: 
lut_result <- lfcShrink(deseq_lut, 
                    contrast=c("condition", "rnai", "control"),
                    res=lut_result)
plotMA(lut_result, ylim=c(-6,6))
```

Extract significantly DE genes with p-value adjusted under multiple test correction using the BH method, and finally the visualization: 
 
```{r, echo=FALSE}
# multiple test correction: (BH method), p-adjusted
summary(lut_result)
lut_all <- data.frame(S1$V1,lut_result)
View(lut_all)
# arrange by more significant DE
lut_sig <- subset(lut_all, padj <.05)
lut_sig <- lut_sig %>%  arrange(padj)
lut_sig_res <- lut_sig[,-1]
rownames(lut_sig_res) <- lut_sig[,1]
```
 - plot the expression heatmap: differences between outgroup and variegatus, between treatment and control
```{r, echo=FALSE, include=TRUE, results="hide"}
# Load library for RColorBrewer
library(RColorBrewer)

# heat map
sig_norm_counts_lut <- lut_normalized_counts[rownames(lut_sig_res), ]
heat_colors <- brewer.pal(n = 6, name = "YlGnBu")
pheatmap(sig_norm_counts_lut, 
         color = heat_colors, 
         cluster_rows = T, 
         show_rownames = F,
         annotation = select(lut_meta, condition), 
         scale = "row")
```
 - Volcano plot: it looks like there are a few more strongly up-regulated genes after RNAi than down-regulated: 
```{r, echo=FALSE, include=TRUE, results="hide"}
# Obtain logical vector regarding whether padj values are less than 0.05  
lut_logic_all <- data.frame(lut_all) %>% mutate(threshold = padj < 0.05)

# Volcano plot
# Create the volcano plot
ggplot(lut_logic_all) + 
  geom_point(aes(x = log2FoldChange, y = -log10(padj), color = threshold)) + 
  xlab("log2 fold change") + 
  ylab("-log10 adjusted p-value") + 
  theme(legend.position = "none", 
        plot.title = element_text(size = rel(1.5), hjust = 0.5), 
        axis.title = element_text(size = rel(1.25)))
```


Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
